pkg/latte/intent_verify_login_link.go

```diff
@@ -0,0 +1,79 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentVerifyLoginLink{})
+}
+
+var IntentVerifyLoginLinkSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false,
+		"properties": {
+			"code": { "type": "string" }
+		},
+		"required": ["code"]
+	}
+`)
+
+type IntentVerifyLoginLink struct {
+	Code string `json:"code"`
+}
+
+func (i *IntentVerifyLoginLink) Kind() string {
+	return "latte.IntentVerifyLoginLink"
+}
+
+func (i *IntentVerifyLoginLink) JSONSchema() *validation.SimpleSchema {
+	return IntentVerifyLoginLinkSchema
+}
+
+func (i *IntentVerifyLoginLink) GetCode() string {
+	return i.Code
+}
+
+func (i *IntentVerifyLoginLink) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	return []workflow.Input{}, nil
+}
+
+func (i *IntentVerifyLoginLink) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	codeModel, err := deps.OTPCodeService.SetUserInputtedMagicLinkCode(i.GetCode())
```

Comment:
I do it in intent because this is not an official feature of authgear. Or would still prefer to do it in node instead?